syntax = "proto3";

package poohbear;

message Trade {
  string pair = 1;
  float rate = 2;
  float amount = 3;
  float total = 4;
  string timestamp = 5;
}

message TradeBlock {
  string pair = 1;
  string timestamp = 2;
  repeated Trade trades = 3;
}

message TradeBlockRange {
  string pair = 1;
  repeated TradeBlock trades = 2;
}

message DateRange {
  string pair = 1;
  string start = 2;
  string end = 3;
}


service TradeRange {
  rpc GetTradeRange(DateRange) returns (TradeBlockRange) {}
}
